clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2958,47 +2983,171 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
   return Result;
 }
 
-// Handles reqd_work_group_size, work_group_size_hint and max_work_group_size
+template <typename WorkGroupAttrType>
+void Sema::addIntelSYCLTripleArgFunctionAttr(Decl *D,
+                                             const AttributeCommonInfo &CI,
+                                             Expr *XDim, Expr *YDim,
+                                             Expr *ZDim) {
+  assert(XDim && "Attribute must have an argument.");
+  assert(YDim && "Attribute must have an argument.");
+  assert(ZDim && "Attribute must have an argument.");
+
+  if (!XDim->isInstantiationDependent() || !YDim->isInstantiationDependent() ||
```

Comment:
I think this logic should be using `&&` and not `||` -- just because one dimension isn't dependent doesn't mean that the others aren't. That would be a good test case, btw.

Reply:
Done